R/GridPrediction.R

```diff
@@ -0,0 +1,80 @@
+#' @include Grid.R
+#' @include generics.R
+NULL
+
+
+#' @rdname Grid-Dev
+.GridPrediction <- setClass(
+    "GridPrediction",
+    contains = "Grid",
+    slots = c(
+        "times" = "numeric_or_NULL",
+        "newdata" = "data.frame",
+        "params" = "list"
+    )
+)
+#' @rdname Grid-Functions
+#' @export
+GridPrediction <- function(times = NULL, newdata, params = NULL) {
+    .GridPrediction(
+        times = times,
+        params = params,
+        newdata = newdata
+    )
+}
+
+#' @rdname Quant-Dev
+#' @export
+as.QuantityGenerator.GridPrediction <- function(object, data, ...) {
+
+    assert_class(data, "DataJoint")
+    data_list <- as.list(data)
+    validate_time_grid(object@times)
+
+    n_times <- length(object@times)
+    n_obs <- nrow(object@newdata)
+    newdata <- object@newdata
```

Comment:
unlikely to happen, but maybe safer to add an assertion here that this column name does not exist yet?